### PR TITLE
Doma-1264 Domain acquiring (Proof of concept)

### PR DESCRIPTION
**Pull request by SavelevMatthew**
Created base schemas for acquiring domain, but removed all validations, since flow changed dramatically few days go.

# Important note
This task will be changed during next 1-2 week, for now it's need to remove block from all members of billing team. So...don't expect perfect code here.

### Based on https://github.com/open-condo-software/condo/pull/688 PR

### CODE REVIEW

File: apps/condo/domains/acquiring/access/AcquiringIntegration.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Generated by `createschema acquiring.AcquiringIntegration 'name:Text;'`
+ */
+
+const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')
+
+async function canReadAcquiringIntegrations ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
```
Comment (pahaz) [inline apps/condo/domains/acquiring/access/AcquiringIntegration.js:9]: wow! we need it everywhere! %)

Comment (pahaz) [reply apps/condo/domains/acquiring/access/AcquiringIntegration.js:9]: what about isAdmin return {} ?

File: apps/condo/domains/acquiring/access/AcquiringIntegrationContext.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin) return true
+    let organizationId
+    let integrationId
+    if (operation === 'create') {
+        // get ids from input on create
+        organizationId = get(originalInput, ['organization', 'connect', 'id'])
+        integrationId = get(originalInput, ['integration', 'connect', 'id'])
+        if (!organizationId || !integrationId) return
```
Comment (pahaz) [inline apps/condo/domains/acquiring/access/AcquiringIntegrationContext.js:34]: return false

File: apps/condo/domains/acquiring/access/AcquiringIntegrationContext.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        // getting ids from existing object
+        if (!itemId) return false
+        const context = await getById('AcquiringIntegrationContext', itemId)
+        if (!context) return false
+        const { organization, integration } = context
+        organizationId = organization
+        integrationId = integration
+    }
+    const canManageIntegrations = await checkOrganizationPermission(context, user.id, organizationId, 'canManageIntegrations')
+    if (canManageIntegrations && operation === 'create') return true
```
Comment (pahaz) [inline apps/condo/domains/acquiring/access/AcquiringIntegrationContext.js:45]: organization admin can only create? what about soft delete? or update?

File: apps/condo/domains/acquiring/schema/MultiPayment.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        },
+
+        implicitFee: {
+            schemaDoc: 'Total amount of money charged from recipients of multipayment as fee for transaction',
+            type: Decimal,
+            isRequired: false,
+        },
+
+        // TODO (savelevMatthew): move to MONEY_FIELD later
+        amountWithOutExplicitFee: {
```
Comment (toplenboren) [inline apps/condo/domains/acquiring/schema/MultiPayment.js:60]: WIthout is a single word

File: apps/condo/domains/acquiring/schema/Payment.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            schemaDoc: 'Link to a billing receipt that the user paid for',
+            type: Relationship,
+            ref: 'BillingReceipt',
+            isRequired: true,
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.PROTECT' },
+        },
+
+        frozenReceipt: {
+            schemaDoc: 'Frozen billing receipt, used to resolving future conflicts',
```
Comment (toplenboren) [inline apps/condo/domains/acquiring/schema/Payment.js:70]:
```suggestion
            schemaDoc: 'Frozen billing receipt, used to resolve conflicts',
```

**Review by toplenboren — APPROVED** [review]

File: apps/condo/domains/acquiring/access/Payment.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    if (user.deletedAt) return false
+    if (user.isAdmin) return true
+    if (operation === 'create') {
+        // Nobody can create Payments manually
+        return false
+
+    } else if (operation === 'update') {
+        if (!itemId) return false
+        // Acquiring integration can update it's own Payments
+        const payments = await Payment.getAll(context, {
```
Comment (pahaz) [inline apps/condo/domains/acquiring/access/Payment.js:40]: looks strange!can we use something like `integration: { accessRights_some: { user: { id: user.id } } } }`

File: apps/condo/domains/acquiring/access/AcquiringIntegration.js (hunk 1 above)
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/access/AcquiringIntegration.js:9]: Everybody should be able to access this schema (may be will do field accesses later)

File: apps/condo/domains/acquiring/access/AcquiringIntegrationContext.js (hunk 3 above)
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/access/AcquiringIntegrationContext.js:45]: I guess we should talk about that...Right now it' working like a billing's ones, but I guess here can be different cases

File: apps/condo/domains/acquiring/constants.js [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+const PAYMENT_ERROR_STATUS = 'ERROR'
+
+const PAYMENT_STATUSES = [
+    PAYMENT_INIT_STATUS,
+    PAYMENT_PROCESSING_STATUS,
+    PAYMENT_DONE_STATUS,
+    PAYMENT_ERROR_STATUS,
+]
+
+const AMOUNT_PRECISION_DIFFERENCE = Math.pow(10, -4)
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/constants.js:116]: I guess we should fix it later (https://doma.atlassian.net/browse/DOMA-1445)

File: apps/condo/domains/acquiring/schema/MultiPayment.js (hunk 4 above)
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/schema/MultiPayment.js]: changed 👍

File: apps/condo/domains/acquiring/schema/Payment.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+            schemaDoc: 'Time at which money was advanced to recipient\'s account',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+
+        accountNumber: {
+            schemaDoc: 'Payer\'s account number',
+            type: Text,
+            isRequired: true,
+        },
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/schema/Payment.js:52]: Billing account number contains not normalised Text, so there'll be a lot of mismatches...So the answer is no...at least for now

File: apps/condo/domains/acquiring/schema/Payment.js (hunk 5 above)
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/schema/Payment.js:70]: fixed 👍

File: apps/condo/domains/acquiring/access/MultiPayment.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+async function canManageMultiPayments ({ authentication: { item: user }, operation, itemId, context }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin) return true
+    // Can be created only through custom mutation or by admin, can be modified by acquiring integration account
+    if (operation === 'create') {
+        return false
+    } else if (operation === 'update') {
+        if (!itemId) return false
+        const availableIntegrations = await AcquiringIntegration.getAll(context, {
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/access/MultiPayment.js:38]: Can you describe it with more details? Can't understand what're you asking for

[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/access/MultiPayment.js:38]: Figured out, why did you mean...I'll fix that :)

File: apps/condo/domains/acquiring/constants.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/constants.js:116]: Moved to [big.js](https://github.com/MikeMcl/big.js/) for decimal calculation / comparison

File: apps/condo/domains/acquiring/schema/AcquiringIntegration.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+            isRequired: true,
+        },
+
+        accessRights: {
+            type: Relationship,
+            ref: 'AcquiringIntegrationAccessRight.integration',
+            many: true,
+        },
+
+        canGroupReceipts: {
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/acquiring/schema/AcquiringIntegration.js:30]: Added check inside `RegisterMultiPayment` mutation

**Pull request merged**